src/app/git.service.ts

```diff
@@ -0,0 +1,68 @@
+var html2md = require("html2markdown");
+var path = require("path");
+var promisify = require("promisify-node");
+var fse = promisify(require("fs-extra"));
+fse.ensureDir = promisify(fse.ensureDir);
+const git = require('simple-git/promise');
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class gitService {
+  constructor(private repoPath: string, private simpleGit) {
+  }
+
+  SetRepo(repoPath:string) {
+        this.repoPath = repoPath;
+        this.simpleGit = require('simple-git')(this.repoPath);
+  }
+
+  Clone(remote:string) {
+       git().silent(true)
+           .clone(remote)
+           .then(() => console.log('finished'))
+           .catch((err) => console.error('failed: ', err));
+  }
+
+  Checkout(branch_name:string) {
+        this.simpleGit.checkout(branch_name);
+  }
+
+  CreateBranch(branch_name:string) {
+      this.simpleGit.checkoutLocalBranch(branch_name);
+  }
+
+  Commit() {
+      var message:string = "Write Jekyll blog";
+      this.simpleGit.add("./*").commit(message);
+  }
+
+  Pull(remote:string, remote_branch:string) {
+      this.simpleGit.pull(remote, remote_branch);
+  }
+
+  Push(remote:string, remote_branch:string){
+      this.simpleGit.push(remote, remote_branch);
+  }
+
+  HeaderHtml2md(header:any){
```

Comment:
I think we should separate html2md into new module.
Git service class should do something related in 'git'